### ISSUE TITLE
Regression in 0.3.0: The same string is interned differently

### ISSUE THREAD

**Issue by connorskees** [Kixiron/lasso#7]
Hello,

I've recently come across a very interesting case that seems to cause the interner to give different indices for the same string interned twice. 

This is a regression from `0.2.0` to `0.3.0`.

I have reduced the case to

```rust
use lasso::Rodeo;

fn main() {
    let mut rodeo = Rodeo::default();

    rodeo.get_or_intern("a");
    rodeo.get_or_intern("b");
    rodeo.get_or_intern("c");
    rodeo.get_or_intern("d");
    rodeo.get_or_intern("e");
    rodeo.get_or_intern("f");
    rodeo.get_or_intern("g");
    rodeo.get_or_intern("h");
    rodeo.get_or_intern("i");
    rodeo.get_or_intern("j");
    rodeo.get_or_intern("k");
    rodeo.get_or_intern("l");
    rodeo.get_or_intern("m");
    rodeo.get_or_intern("n");
    rodeo.get_or_intern("o");
    rodeo.get_or_intern("p");
    rodeo.get_or_intern("q");
    rodeo.get_or_intern("r");
    rodeo.get_or_intern("s");
    rodeo.get_or_intern("t");
    rodeo.get_or_intern("u");
    rodeo.get_or_intern("v");
    rodeo.get_or_intern("w");
    rodeo.get_or_intern("x");
    rodeo.get_or_intern("y");
    rodeo.get_or_intern("z");
    rodeo.get_or_intern("aa");
    rodeo.get_or_intern("bb");
    rodeo.get_or_intern("cc");
    rodeo.get_or_intern("dd");
    rodeo.get_or_intern("ee");
    rodeo.get_or_intern("ff");
    rodeo.get_or_intern("gg");
    rodeo.get_or_intern("hh");
    rodeo.get_or_intern("ii");
    rodeo.get_or_intern("jj");
    rodeo.get_or_intern("kk");
    rodeo.get_or_intern("ll");
    rodeo.get_or_intern("mm");
    rodeo.get_or_intern("nn");
    rodeo.get_or_intern("oo");
    rodeo.get_or_intern("pp");
    rodeo.get_or_intern("qq");
    rodeo.get_or_intern("rr");
    rodeo.get_or_intern("ss");
    rodeo.get_or_intern("tt");
    rodeo.get_or_intern("uu");
    rodeo.get_or_intern("vv");
    rodeo.get_or_intern("ww");
    rodeo.get_or_intern("xx");
    rodeo.get_or_intern("yy");
    rodeo.get_or_intern("zz");
    rodeo.get_or_intern("aaa");
    rodeo.get_or_intern("bbb");
    rodeo.get_or_intern("ccc");

    let var = rodeo.get_or_intern("ddd");

    rodeo.get_or_intern("eee");

    dbg!(var, rodeo.get_or_intern("ddd"));
}
```

```toml
[dependencies]
lasso = "0.3.0"
```

The output that I see from this is

```
[src/main.rs:66] var = Spur {
    key: 56,
}
[src/main.rs:66] rodeo.get_or_intern("ddd") = Spur {
    key: 58,
}
```

**Comment by Kixiron** [comment]
Thank you so much for reporting this! It turns out there was a flaw in my [insert with hasher](https://github.com/Kixiron/lasso/blob/master/src/single_threaded.rs#L245-L252) functions, but I've fixed it and added a [regression test](https://github.com/Kixiron/lasso/blob/master/src/single_threaded.rs#L916), can you point yourself at the latest git and make sure it works on your side before I publish a patch?

**Comment by connorskees** [comment]
All my unit tests (including one that triggered this bug) pass with `lasso = { git = "https://github.com/Kixiron/lasso" }`. Thanks for the quick fix!

**Comment by Kixiron** [comment]
Released `lasso v0.3.1` with the patch, thank you again!
